### ISSUE TITLE
Use single quotes in no-undef context.report message

### ISSUE THREAD

**Issue by lencioni** [eslint/eslint#4845]
While working on a different project that consumes the output of ESLint
and some plugins, [we noticed that the quote styles emitted by the
no-undef rule was different than the quote styles emitted by some of the
rules in our other plugins](https://github.com/trotzig/import-js/issues/85#issuecomment-168332322). This inconsistency led to a small bug in
our plugin, so I wanted to smooth things out a bit for future
developers.

To make things more consistent, my first inclination was to change the
plugin to match ESLint's no-undef quote style, but upon investigation it
seems that most of the core ESLint rules use single quotes instead of
double quotes, so I think it makes sense to change this one to be more
consistent.

There may be other inconsistencies that are worth resolving, but #4840 
fixes the main pain point for me.

**Comment by eslintbot** [comment]
@lencioni Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by lencioni** [comment]
> The version of ESLint you are using (run eslint -v)

v1.10.3

> What you did (the source code and ESLint configuration)

`.eslintrc.js`:

```js
module.exports = {
  "env": {
    "browser": true
  },
  "rules": {
    "no-undef": 2
  }
};
```

`index.js`:

```js
console.log(thisIsUndef);
```

```sh
eslint index.js
```
> The actual ESLint output complete with numbers

```
~/eslint-example ❯❯❯ eslint index.js

/Users/joe/eslint-example/index.js
  1:13  error  "thisIsUndef" is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

> What you expected to happen instead

```
~/eslint-example ❯❯❯ eslint index.js

/Users/joe/eslint-example/index.js
  1:13  error  'thisIsUndef' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by michaelficarra** [comment]
While it would be good to have consistency here, your motivation worries me. The pretty formatter is meant to be consumed by humans. If a program you are writing is consuming the output of eslint, you should be using a formatter that is creating output that is meant to be consumed in this way.

**Comment by lencioni** [comment]
Sorry, I should have specified. We are actually using the [compact formatter](https://github.com/trotzig/import-js/blob/d08920a4/lib/import_js/importer.rb#L106-L108). I'm looking over [the formatters that are available](http://eslint.org/docs/user-guide/formatters/) and thinking that json or unix would perhaps be a little better for us, but wouldn't resolve this problem for us.

**Comment by lencioni** [comment]
I just noticed that the other rule we are consuming, no-unused-vars, also uses double quotes instead of single quotes, so I'll add another commit to this PR to change that.

**Comment by mysticatea** [comment]
I think ESLint rules are using double quotes consistently.

**Comment by nzakas** [comment]
@mysticatea he means inside strings, such as "Missing 'foo'". This is not covered by the rule.

Before we go any further on this, I'd like to see a breakdown of rules using double vs. single quotes to make sure single is actually the predominant one, otherwise we will just keep converting things back and forth. @lencioni can you do that?

**Comment by lencioni** [comment]
Sure, I'd be happy to! I'll probably dig into this tomorrow.

**Comment by lencioni** [comment]
I ran `ag context.report.*{{ -- lib/rules` after my commit that converts 2 rules to singles and got the following output. Although this isn't a perfect method, it seems close enough for our needs.

```
lib/rules/camelcase.js:38:        context.report(node, "Identifier '{{name}}' is not in camel case.", { name: node.name });
lib/rules/complexity.js:50:            context.report(node, "Function '{{name}}' has a complexity of {{complexity}}.", { name: name, complexity: complexity });
lib/rules/id-match.js:60:        context.report(node, "Identifier '{{name}}' does not match the pattern '{{pattern}}'.", {
lib/rules/jsx-quotes.js:53:                context.report(attributeValue, "Unexpected usage of {{description}}.", setting);
lib/rules/max-depth.js:50:            context.report(node, "Blocks are nested too deeply ({{depth}}).",
lib/rules/max-nested-callbacks.js:42:            context.report(node, "Too many nested callbacks ({{num}}). Maximum allowed is {{max}}.", opts);
lib/rules/max-params.js:26:            context.report(node, "This function has too many parameters ({{count}}). Maximum allowed is {{max}}.", {
lib/rules/no-alert.js:30:    context.report(node, "Unexpected {{name}}.", { name: identifierName });
lib/rules/no-arrow-condition.js:64:            context.report(node, "Arrow function `=>` used inside {{statementType}} instead of comparison operator.", {statementType: node.type});
lib/rules/no-bitwise.js:31:        context.report(node, "Unexpected use of '{{operator}}'.", { operator: node.operator });
lib/rules/no-caller.js:21:                context.report(node, "Avoid arguments.{{property}}.", { property: propertyName });
lib/rules/no-catch-shadow.js:50:                context.report(node, "Value of '{{name}}' may be overwritten in IE 8 and earlier.",
lib/rules/no-cond-assign.js:108:            context.report(ancestor, "Unexpected assignment within {{type}}.", {
lib/rules/no-dupe-class-members.js:76:                context.report(node, "Duplicate name \"{{name}}\".", {name: name});
lib/rules/no-dupe-keys.js:36:                        context.report(node, property.loc.start, "Duplicate key '{{key}}'.", { key: keyName });
lib/rules/no-empty-label.js:20:                context.report(node, "Unexpected label \"{{l}}\"", {l: node.label.name});
lib/rules/no-inner-declarations.js:50:            context.report(node, "Move {{type}} declaration to {{body}} root.",
lib/rules/no-obj-calls.js:20:                    context.report(node, "'{{name}}' is not a function.", { name: name });
lib/rules/no-octal-escape.js:29:                    context.report(node, "Don't use octal: '\\{{octalDigit}}'. Use '\\u....' instead.",
lib/rules/no-new-wrappers.js:19:                context.report(node, "Do not use {{fn}} as a constructor.", { fn: node.callee.name });
lib/rules/no-restricted-imports.js:28:                    context.report(node, "'{{importName}}' import is restricted from being used.", {
lib/rules/no-restricted-modules.js:65:                    context.report(node, "'{{moduleName}}' module is restricted from being used.", {
lib/rules/no-restricted-syntax.js:21:        context.report(node, "Using \"{{type}}\" is not allowed.", node);
lib/rules/no-undef-init.js:23:                context.report(node, "It's not necessary to initialize '{{name}}' to undefined.", { name: name });
lib/rules/prefer-reflect.js:47:        context.report(node, "Avoid using {{existing}}, instead use {{substitute}}", {
lib/rules/quote-props.js:146:                context.report(node, "Properties should be quoted as `{{property}}` is a reserved word.", {property: key.name});
lib/rules/quote-props.js:152:                context.report(node, "Inconsistently quoted property `{{key}}` found.", {
lib/rules/valid-jsdoc.js:127:                            context.report(jsdocNode, "Missing JSDoc parameter type for '{{name}}'.", { name: tag.name });
lib/rules/valid-jsdoc.js:131:                            context.report(jsdocNode, "Missing JSDoc parameter description for '{{name}}'.", { name: tag.name });
lib/rules/valid-jsdoc.js:135:                            context.report(jsdocNode, "Duplicate JSDoc parameter '{{name}}'.", { name: tag.name });
lib/rules/valid-jsdoc.js:174:                    context.report(jsdocNode, "Use @{{name}} instead.", { name: prefer[tag.title] });
lib/rules/valid-jsdoc.js:196:                            context.report(jsdocNode, "Expected JSDoc for '{{name}}' but found '{{jsdocName}}'.", {
lib/rules/valid-jsdoc.js:201:                            context.report(jsdocNode, "Missing JSDoc for parameter '{{name}}'.", {
```

It looks like a bunch of rules don't use quotes, 2 use backticks (quote-props, no-arrow-condition), 3 use double quotes (no-dupe-class-members, no-empty-label, no-restricted-syntax), and 12 use single quotes (camelcase, complexity, id-match, no-bitwise, no-catch-shadow, no-dupe-keys, no-obj-calls, no-octal-escape, no-restricted-imports, no-restricted-modules, no-undef-init, valid-jsdoc).

I'd be happy to submit changes to make more things consistent if you are open to that.

**Comment by nzakas** [comment]
Nice work! Yeah, I'd love a PR to make things consistent. Keep in mind that some messages use string concatenation, so there are likely more outliers in the codebase.

**Comment by lencioni** [comment]
Cool. I'll update my PR shortly.

**Comment by lencioni** [comment]
@nzakas You were right, I did find more outliers. I have pushed an updated commit that changes all of the ones that I found by searching for `context.report` and verifying them manually.
